### ISSUE TITLE
Craftsman Door Opener not working in 3.1.0

### ISSUE THREAD

**Issue by justintime** [brbeaird/SmartThings_MyQ#71]
My ST integration with my Craftsman doors stopped working with Chamberlain pushed the v5 API update.  After updating to the 3.1.0 version of the smartapp and DH, I'm not able to login to my account from the app.

When I first bought the openers, I thought that I could use the Chamberlain MyQ iOS app on my phone to control them, so I ended up setting up an account on MyQ.  That account has 0 devices on it.

I later set up my account in Assurelink, and associated my openers in that account.

The important thing here is that I used different passwords on those two different accounts.

In the new smartapp, I'm unable to login with my Assurelink password.  It tells me the username or password is incorrect.  However, if I use my MyQ password, it allows me to login.  However, since I'm using the wrong account on the wrong gateway, the app doesn't see that I have any openers available.

Was there some changes to the code where the gateway detection logic was changed?

**Comment by brbeaird** [comment]
> When I first bought the openers, I thought that I could use the Chamberlain MyQ iOS app on my phone to control them, so I ended up setting up an account on MyQ. That account has 0 devices on it.

I think this is the root of the problem. I saw somewhere else where someone mentioned people with Craftsman accounts needed to take steps to migrate their account to MyQ as the other site is shutting down soon. I'm not sure what's involved with that, but unless you somehow get the devices showing up in the native MyQ app, they won't show up in this app.

**Comment by justintime** [comment]
Yeah, it looks like the Craftsman gateway is gone in the current version.  The prior version would choose between two API endpoints: https://github.com/brbeaird/SmartThings_MyQ/blob/1900ebe45a11fd3f04d8e6b295e09901b0e91e84/smartapps/brbeaird/myq-lite.src/myq-lite.groovy#L1098

Now there's just one, and it's a MyQ endpoint.

**Comment by justintime** [comment]
Hmm, I'll research and see if there's any way for me to add my devices to MyQ.  A couple years ago it wasn't possible.

**Comment by brbeaird** [comment]
Yes, I consolidated them based on two reports that the old ones had stopped working already.

Here's one reference where someone mentioned they successfully converted:
https://community.smartthings.com/t/beta-myq-lite-3-0-need-testers/174608/17?u=brbeaird

**Comment by justintime** [comment]
I was able to do the following to get things set up in MyQ:

1. Remove all devices from the Assurelink at https://assurelink.craftsman.com/Home
2. Create a new account in the MyQ app
3. Add my serial numbers to the MyQ app

This did result in having to reset all my automations, but at least I'm hopefully futureproof for a while at this point.  I didn't test to see if the old endpoint was still working, but it certainly was before the v5 update -- I debated on hardcoding the variable to the Assurelink endpoint but just decided to bite the bullet and migrate instead.

It's actually a testament to this app that I ended up in this spot.  I never used the Assurelink app to open and close the doors, I just used the ST interface because it was so reliable.  Because of that, I never got any notifications that I needed to migrate.
